monitoring/monitorlib/messagesigning/message_signer.py

```diff
@@ -0,0 +1,117 @@
+import base64
+import time
+from Crypto.PublicKey import RSA
+from Crypto.Signature import pkcs1_15
+from Crypto.Hash import SHA256
+from loguru import logger
+import urllib.parse
+from monitoring.monitorlib.messagesigning.hasher import get_content_digest
+
+
+def get_x_utm_jws_header(cert_url):
+    return '"alg"="{}", "typ"="{}", "kid"="{}", "x5u"="{}"'.format(
+        "RS256",
+        "JOSE",
+        get_key_id(),
+        cert_url,
+    )
+
+
+def get_signed_headers(object_to_sign, private_key_path, cert_url):
+    signed_type = str(type(object_to_sign))
+    is_signing_request = "PreparedRequest" in signed_type
+    sig, sig_input = get_signature(object_to_sign, signed_type, private_key_path)
+    content_digest = (
+        get_content_digest(object_to_sign.body)
+        if is_signing_request
+        else get_content_digest(object_to_sign.get_data())
+    )
+    signed_headers = {
+        "x-utm-message-signature-input": "utm-message-signature={}".format(sig_input),
+        "x-utm-message-signature": "utm-message-signature=:{}:".format(sig),
+        "x-utm-jws-header": get_x_utm_jws_header(cert_url),
+        "content-digest": "sha-512=:{}:".format(content_digest),
+    }
+    return signed_headers
+
+
+def get_signature_input(sig_base):
+    sig_base_comps = sig_base.split("\n")
+    sig_param_str = [
+        item for item in sig_base_comps if "@signature-params" in item
+    ].pop()
+    start_sig_input_ind = sig_param_str.index("(")
+    end_sig_input_str = len(sig_param_str)
+    return sig_param_str[start_sig_input_ind:end_sig_input_str]
+
+
+def get_signature(object_to_sign, signed_type, private_key_path):
+    sig_base = get_signature_base(object_to_sign, signed_type)
+    sig_base_bytes = bytes(sig_base, "utf-8")
+    sig_input = get_signature_input(sig_base)
+    hash = SHA256.new(sig_base_bytes)
+    with open(private_key_path, "rb") as priv_key_file:
+        private_key = RSA.import_key(priv_key_file.read())
+    return (
+        base64.b64encode(pkcs1_15.new(private_key).sign(hash)).decode("utf-8"),
+        sig_input,
+    )
+
+
+def get_key_id():
+    return "mock_uss_priv_key"
```

Comment:
Since there's no logic in this function, I'm putting the hardcoded string in the `get_x_utm_jws_header` function itself. Removing this function altogether